app/components/Settings/ConnectButton/ConnectButton.react.js

```diff
@@ -4,166 +4,57 @@ import classnames from 'classnames';
-        this.testClass = this.testClass.bind(this);
-        this.updateStatus = this.updateStatus.bind(this);
-        this.state = {
-            hover: false
-        };
-    }
-
-    testClass() {
-        /*
-            Return the connection state as class-status.
-            Knowing this status and getting the errorMessage and buttonMessage
-            from their respective className tags will suffice to test this
-            comoponent.
-        */
-        return `test-${this.props.connection.get('status')}`;
-    }
-
-    connect() {
-        this.updateStatus(APP_STATUS.CONNECTING);
-        this.props.sessionsActions.connect();
-    }
-
-    disconnect() {
-        this.props.sessionsActions.disconnect();
-    }
-
-    onMouseOver() {
-        this.setState({hover: true});
-    }
-
-    onMouseOut() {
-        this.setState({hover: false});
-    }
-
-    updateStatus(status) {
-        if (status !== this.props.connection.get('status')) {
-            this.props.sessionsActions.updateConnection({status});
-            this.setState({
-                buttonMessage: BUTTON_MESSAGE[status]
-            });
-        }
-    }
 
-    componentWillReceiveProps(nextProps) {
-        const status = this.props.connection.get('status');
-            if (!nextProps.ipc.get('error')) {
-                if (
-                    nextProps.ipc.has('databases') &&
-                    !nextProps.ipc.get('databases')
-                ) {
-                    this.updateStatus(APP_STATUS.DISCONNECTED);
-                } else if (nextProps.ipc.get('databases')) {
-                    this.updateStatus(APP_STATUS.CONNECTED);
-                }
+    render() {
+        const {
+            connectionsHaveBeenSaved,
+            connect,
+            connectRequest,
+            saveConnectionsRequest,
+            editMode
+        } = this.props;
+
+        let buttonText;
+        let buttonClick = () => {};
+        let error = null;
+
+        if (!connectRequest.status) {
+            buttonText = 'Connect';
+            buttonClick = connect;
+        } else if (connectRequest.status === 'loading') {
+            buttonText = 'Connecting...';
+        } else if (connectRequest.status >= 200 && connectRequest.status < 300) {
+            if (editMode) {
+                buttonText = 'Save changes';
+                buttonClick = connect;
             } else {
-                if (nextProps.ipc.getIn(['error', 'name']) === 'ConnectionError') {
-                    this.updateStatus(APP_STATUS.CON_ERROR);
-                } else if (status !== APP_STATUS.CON_ERROR) {
-                    this.updateStatus(APP_STATUS.ERROR);
-                }
+                buttonText = 'Connected';
             }
+        } else if (connectRequest.status >= 400 || saveConnectionsRequest.status >= 400) {
+           buttonText = 'Connect';
+           buttonClick = connect;
+           // TODO - Try out bad connections to verify this.
```

Comment:
Looks like error messages aren't getting propagated into the front-end
![bad-connection](https://cloud.githubusercontent.com/assets/1280389/20782735/94cec968-b75c-11e6-9d2f-7967e1ee8cc9.gif)